src/hotspot/share/gc/shenandoah/shenandoahSTWMark.cpp

```diff
@@ -148,14 +145,11 @@ void ShenandoahSTWMark::mark_roots(uint worker_id) {
 }
 
 void ShenandoahSTWMark::finish_mark(uint worker_id) {
-  assert(ShenandoahHeap::heap()->gc_generation() == _generation, "Marking unexpected generation");
   ShenandoahPhaseTimings::Phase phase = _full_gc ? ShenandoahPhaseTimings::full_gc_mark : ShenandoahPhaseTimings::degen_gc_stw_mark;
   ShenandoahWorkerTimingsTracker timer(phase, ShenandoahPhaseTimings::ParallelMark, worker_id);
-  ShenandoahReferenceProcessor* rp = _generation->ref_processor();
-  shenandoah_assert_generations_reconciled();
   StringDedup::Requests requests;
 
-  mark_loop(worker_id, &_terminator, rp,
-            _generation->type(), false /* not cancellable */,
+  // TODO: Why are we passing our own fields to our own method?
```

Comment:
It's a method on our parent class which doesn't have a Terminator object handy.

Reply:
I think it's mostly because other uses of this (public) method are from `MarkingTasks` which have a `Terminator` but not a `ShenandoahGeneration`. I deleted the TODO.